python/pyarrow/_parquet.pyx

```diff
@@ -187,86 +175,70 @@ cdef class ColumnChunkMetaData:
-        def __get__(self):
-            return self.metadata.num_values()
-
-    property path_in_schema:
-
-        def __get__(self):
-            path = self.metadata.path_in_schema().get().ToDotString()
-            return frombytes(path)
-
-    property is_stats_set:
-
-        def __get__(self):
-            return self.metadata.is_stats_set()
-
-    property statistics:
-
-        def __get__(self):
-            if not self.metadata.is_stats_set():
-                return None
-            statistics = RowGroupStatistics()
-            statistics.init(self.metadata.statistics())
-            return statistics
-
-    property compression:
-
-        def __get__(self):
-            return self.metadata.compression()
+    @property
+    def file_offset(self):
+        return self.metadata.file_offset()
 
-    property encodings:
+    @property
+    def file_path(self):
+        return frombytes(self.metadata.file_path())
 
-        def __get__(self):
-            return map(encoding_name_from_enum,
-                       self.metadata.encodings())
+    @property
+    def physical_type(self):
+        return physical_type_name_from_enum(self.metadata.type())
 
-    property has_dictionary_page:
+    @property
+    def num_values(self):
+        return self.metadata.num_values()
 
-        def __get__(self):
-            return self.metadata.has_dictionary_page()
+    @property
+    def path_in_schema(self):
+        path = self.metadata.path_in_schema().get().ToDotString()
+        return frombytes(path)
 
-    property dictionary_page_offset:
+    @property
+    def is_stats_set(self):
+        return self.metadata.is_stats_set()
 
-        def __get__(self):
-            return self.metadata.dictionary_page_offset()
+    @property
+    def statistics(self):
+        if not self.metadata.is_stats_set():
+            return None
+        statistics = RowGroupStatistics()
+        statistics.init(self.metadata.statistics())
+        return statistics
 
-    property data_page_offset:
+    @property
+    def compression(self):
+        return compression_name_from_enum(self.metadata.compression())
 
-        def __get__(self):
-            return self.metadata.data_page_offset()
+    @property
+    def encodings(self):
+        return tuple(map(encoding_name_from_enum, self.metadata.encodings()))
 
-    property index_page_offset:
+    @property
+    def has_dictionary_page(self):
+        return bool(self.metadata.has_dictionary_page())
 
-        def __get__(self):
-            return self.metadata.index_page_offset()
+    @property
+    def dictionary_page_offset(self):
+        return self.metadata.dictionary_page_offset()
 
-    property total_compressed_size:
+    @property
+    def data_page_offset(self):
+        return self.metadata.data_page_offset()
 
-        def __get__(self):
-            return self.metadata.total_compressed_size()
+    @property
+    def index_page_offset(self):
+        return self.metadata.index_page_offset()
```

Comment:
Similarly `index_page_offset` is `0`, is it supposed to be zero, or `index_page` is optional too? (I'm not that familiar with parquet specs yet)

Reply:
`index_page` is optional. This should return `None`.

Reply:
How do I know that `index_page` hasn't been set? `index_page_offset == 0 => return None`?

Reply:
@xhochy ping

Reply:
No, this is buggy. We should export it at the moment. Please raise `NotImplementedError` for now. I'll fix the `parquet-cpp` part over at https://issues.apache.org/jira/browse/PARQUET-1358

Reply:
Do You mean:

```py
@property
def has_index_page(self):
    raise NotImplementedError

@property
def index_page_offset(self):
    # leave it as is
    return self.metadata.index_page_offset()
```

or 

```py
@property
def index_page_offset(self):
    raise NotImplementedError
```

?

Reply:
Files not written with `parquet-cpp` will lead to garbage on `index_page_offset` currently as it is mostly not set.

```
@property
def has_index_page(self):
    raise NotImplementedError("not supported in parquet-cpp")

@property
def index_page_offset(self):
    raise NotImplementedError("parquet-cpp doesn't return valid values")
```

Reply:
Ok, Thanks!